akka-docs/src/test/java/jdocs/stream/operators/SourceOrFlow.java

```diff
@@ -489,6 +490,62 @@ void groupByExample() {
     // #groupBy
   }
 
+  void watchTerminationExample() {
+    // #watchTermination
+    List<IntSupplier> elements = Arrays.asList(() -> 1, () -> 2, () -> 3);
+    Source.from(elements)
+        .watchTermination(
+            (prevMatValue, completionStage) -> {
+              completionStage.whenComplete(
+                  (done, exc) -> {
+                    if (done != null)
+                      System.out.println("The stream materialized " + prevMatValue.toString());
+                    else System.out.println(exc.getMessage());
+                  });
+              return prevMatValue;
+            })
+        .runForeach(element -> System.out.println(element.getAsInt()), system);
+
+    /*
+    Prints:
+    1
+    2
+    3
+    The stream materialized NotUsed
+     */
+
+    List<IntSupplier> elementsWithExc =
+        Arrays.asList(
+            () -> 1,
+            () -> 2,
+            () -> {
+              throw new RuntimeException("Boom");
+            },
+            () -> 3);
+    Source.from(elementsWithExc)
+        .watchTermination(
+            (prevMatValue, completionStage) -> {
+              // this function will be run when the stream terminates
+              // the CompletionStage provided as a second parameter indicates whether
+              // the stream completed successfully or failed
+              completionStage.whenComplete(
+                  (done, exc) -> {
+                    if (done != null)
+                      System.out.println("The stream materialized " + prevMatValue.toString());
+                    else System.out.println(exc.getMessage());
+                  });
+              return prevMatValue;
+            })
+        .runForeach(element -> System.out.println(element.getAsInt()), system);
+    /*
+    Prints:
+    1
+    2
+    Boom
```

Comment:
Good with a failure sample as well!

A bit tricky that the failure is triggered in the sink by passing function elements, I think it would be easier to follow with int elements in stream and do a `if (element == 2) throw Boom` in the Sink lambda instead.

Reply:
This also simplifies the creation of the `Source`s. Fixed!